### PR TITLE
Add more compatability APIs.

### PR DESCRIPTION
**Pull request by SparkiDev**
d2i_ECDSA_SIG, i2d_ECDSA_SIG, EVP_DigestVerifyInit,
EVP_DigestVerifyUpdate, EVP_DigestVerifyFinal, EVP_PKEY_id,
PEM_read_bio_PUBKEY

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -27432,6 +27454,58 @@ int wolfSSL_ECDSA_do_verify(const unsigned char *d, int dlen,
...
+}
+
+int wolfSSL_i2d_ECDSA_SIG(const WOLFSSL_ECDSA_SIG *sig, unsigned char **pp)
+{
+    word32 len;
+
+    if (sig == NULL)
+        return 0;
+
+    len = 2 + 2 + mp_leading_bit((mp_int*)sig->r->internal) +
```
Comment (toddouska) [inline src/ssl.c]: Let's document what the 2's mean here.

File: src/ssl.c [hunk 2]
```diff
@@ -28108,6 +28228,12 @@ int wolfSSL_EVP_PKEY_type(int type)
 }
 
 
+int wolfSSL_EVP_PKEY_id(const EVP_PKEY *pkey)
+{
+    return pkey->type;
```
Comment (toddouska) [inline src/ssl.c]: Do we need to handle a null pkey?

File: src/ssl.c (hunk 1 above)
Comment (SparkiDev) [reply src/ssl.c]: Comment put in and better handling.

File: src/ssl.c (hunk 2 above)
Comment (SparkiDev) [reply src/ssl.c]: OpenSSL doesn't but can't hurt!

**Pull request merged**